### PR TITLE
pause and resume mirror buttons, along with state reflection

### PR DESCRIPTION
**Pull request by heavycrystal**
not the most aligned button

### CODE REVIEW

File: ui/app/mirrors/edit/[mirrorId]/cdcDetails.tsx [hunk 1]
```diff
@@ -140,4 +146,78 @@ export function numberWithCommas(x: any): string {
...
 }
 
+function statusChangeHandle(
+  mirrorConfig: FlowConnectionConfigs,
+  mirrorStatus: FlowStatus
+) {
+  // hopefully there's a better way to do this cast
+  if (mirrorStatus.toString() === FlowStatus[FlowStatus.STATUS_RUNNING]) {
+    return (
+      <button
```
Comment (iskakaushik) [inline ui/app/mirrors/edit/[mirrorId]/cdcDetails.tsx:156]: use these instead
```
import { Button } from '@/lib/Button';
import { Icon } from '@/lib/Icon';
```

See `mirrors/create/page.tsx` as an example of Icon button.

File: ui/app/mirrors/edit/[mirrorId]/cdcDetails.tsx [hunk 2]
```diff
@@ -140,4 +146,78 @@ export function numberWithCommas(x: any): string {
...
+        className='IconButton'
+        aria-label='Pause'
+        onClick={async () => {
```
Comment (iskakaushik) [inline ui/app/mirrors/edit/[mirrorId]/cdcDetails.tsx:159]: can `onClick` have `async` functions? cc: @pankaj-peerdb

Comment (Amogh-Bharadwaj) [reply ui/app/mirrors/edit/[mirrorId]/cdcDetails.tsx:159]: yup they can

File: ui/app/mirrors/edit/[mirrorId]/cdcDetails.tsx [hunk 3]
```diff
@@ -140,4 +146,78 @@ export function numberWithCommas(x: any): string {
...
+          const req: FlowStateChangeRequest = {
+            flowJobName: mirrorConfig.flowJobName,
+            sourcePeer: mirrorConfig.source,
+            destinationPeer: mirrorConfig.destination,
+            requestedFlowState: FlowStatus.STATUS_PAUSED,
+          };
+          await fetch(`/api/mirrors/state_change`, {
+            method: 'POST',
+            body: JSON.stringify(req),
```
Comment (Amogh-Bharadwaj) [inline ui/app/mirrors/edit/[mirrorId]/cdcDetails.tsx:168]: Let's add `cache:'no-store'`

Comment (heavycrystal) [reply ui/app/mirrors/edit/[mirrorId]/cdcDetails.tsx:169]: done

**Review by iskakaushik — APPROVED** [review]

**Pull request merged**